cli: add tests for running against absent server

tbg opened the pull request:
The caveat here is that those might be flaky if another process
opens the port during stressrace, but we will see.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9938)

<!-- Reviewable:end -->

bdarnell commented:
LGTM. I think you could also test this by trying to connect to port 0, which is guaranteed to fail.

tbg committed:
cli: add tests for running against absent server

The caveat here is that those might be flaky if another process
opens the port during stressrace, but we will see.

tschottdorf commented:
Good idea. It changes the error messages though, so I added a comment for any future reader looking at this test if it's flaky.

tamird commented on pkg/cli/cli_test.go:
```diff
@@ -205,7 +205,41 @@ func TestQuit(t *testing.T) {
+		cmd, expOutPattern string
+	}{
+		{`debug kv inc a`,
+			`increment failed: roachpb.Batch RPC failed: rpc error: code = 14 desc = grpc: the connection is unavailable`},
```
don't you want the "grpc wrap" of this error?

tschottdorf replied:
Good point. #9944.
